plugins/gradle/src/org/jetbrains/plugins/gradle/service/task/GradleTaskManager.java

```diff
@@ -102,10 +103,16 @@ public void executeTasks(final @NotNull ExternalSystemTaskId id,
         CancellationToken cancellationToken = cancellationTokenSource.token();
         myHelper.ensureInstalledWrapper(id, rootProjectPath, effectiveSettings, listener, cancellationToken);
       }
-      myHelper.execute(projectPath, effectiveSettings, id, listener, cancellationTokenSource, connection -> {
-        executeTasks(id, tasks, projectPath, effectiveSettings, jvmParametersSetup, listener, connection, cancellationTokenSource);
-        return null;
-      });
+
+      GradleProgressIndicatorWrapper progressIndicatorWrapper = new GradleProgressIndicatorWrapper(id, effectiveSettings, listener);
+      progressIndicatorWrapper.runWithProgressIndicator(
```

Comment:
Instead of wrapping the whole thing, you can simply register the `listenerWithProgressIndicator` (created by GradleProgressIndicatorWrapper) using the following code:
```
    ExternalSystemProgressNotificationManager progressManager =
      ApplicationManager.getApplication().getService(ExternalSystemProgressNotificationManager.class);
      progressManager.addNotificationListener(id, listenerWithProgressIndicator);
```
It will be removed automatically, no need to unsubscribe.